### PR TITLE
[Console] SymfonyStyle: fix block rpadding when escaping '<'

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

screenshot 2015-05-10 a 20 37 46

### CODE REVIEW

**Comment by xabbuh** [comment]
The result looks good. Though I don't understand why this is necessary.

@ogizanagi Did you find the cause for this issue?

**Comment by ogizanagi** [comment]
@xabbuh : AFAIU, it's only du to the extra backslashes introduced by the `OutputFormatter::escape()` method: https://github.com/symfony/symfony/blob/1d45ca894b76c419d212c451c7d0026f9f14442a/src/Symfony/Component/Console/Formatter/OutputFormatter.php#L36

**Comment by xabbuh** [comment]
@ogizanagi Indeed, that makes sense.

:+1:

**Comment by aitboudad** [comment]
:+1:

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 1]
```diff
@@ -85,7 +85,7 @@ public function block($messages, $type = null, $style = null, $prefix = ' ', $pa
 
         foreach ($lines as &$line) {
             $line = sprintf('%s%s', $prefix, $line);
-            $line .= str_repeat(' ', $this->lineLength - Helper::strlen($line));
+            $line .= str_repeat(' ', $this->lineLength - Helper::strlen($line) + substr_count($line, '\<'));
```
Comment (aitboudad) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: would you mind add a comment above that explain the change :) ?

Comment (ogizanagi) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: Should I link this PR as reference ? comment + link or just link ?

Comment (ogizanagi) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: https://github.com/symfony/symfony/pull/14600/files#diff-18b76c26b226bea37b2b821a5e4962ffR88

Let me know what comment you expect if you think it's better :)

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 2]
```diff
@@ -85,7 +85,8 @@ public function block($messages, $type = null, $style = null, $prefix = ' ', $pa
 
         foreach ($lines as &$line) {
             $line = sprintf('%s%s', $prefix, $line);
-            $line .= str_repeat(' ', $this->lineLength - Helper::strlen($line));
+            //https://github.com/symfony/symfony/pull/14600
+            $line .= str_repeat(' ', $this->lineLength - Helper::strlen($line) + substr_count($line, '\<'));
```
Comment (aitboudad) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: use `strlenWithoutDecoration` instead:
```php
$line .= str_repeat(' ', $this->lineLength - Helper::strlenWithoutDecoration($this->getFormatter(), $line));
```

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 3]
```diff
@@ -85,7 +85,8 @@ public function block($messages, $type = null, $style = null, $prefix = ' ', $pa
 
         foreach ($lines as &$line) {
             $line = sprintf('%s%s', $prefix, $line);
-            $line .= str_repeat(' ', $this->lineLength - Helper::strlen($line));
+            //https://github.com/symfony/symfony/pull/14600
```
Comment (aitboudad) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: you can remove this comment now

File: src/Symfony/Component/Console/Style/SymfonyStyle.php (hunk 2 above)
Comment (ogizanagi) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: Actually, it's a good catch, I didn't notice this method. :smiley: 
But, is anything else than this [line](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Console/Formatter/OutputFormatter.php#L182) useful in there ? (The method is called in `Helper::strlenWithoutDecoration`).

Does it justify the overhead if it isn't the case ?

Comment (aitboudad) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: not only the line you mentioned is needed here but it's depend on the style that applied too.

Comment (ogizanagi) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: Alright. It's done.

In fact I was saying this line was the only line required in our case. But, indeed, that's not true. 
I confirm this is needed, if using tags related to a style in blocks.

First my suggestion, second yours:
screenshot 2015-05-13 a 00 30 48


Thank you !

**Comment by nicolas-grekas** [comment]
Thank you @ogizanagi.

**Pull request merged**